# SSMS Package (simplified metagenomic workflow)

Custom workflow(s) built in collaboration with the [Pathogen & Microbiome Institute (PMI)](https://in.nau.edu/pmi/) and
others.

> A prospective bioinformatics project lead by [George Testo](https://github.com/metro1102); started as a part of
> the [2021 Helios Program](https://www.tgen.org/education/helios-scholars-at-tgen/).

![Imgur](https://i.imgur.com/DNQ0idS.png)

## Project Contributions

Before contributing, please consider reading the following guidelines. We would like to highlight the importance of git
workflow eitquette. For example, commit messages should be consistent with whatever is being added, removed, or changed.
Please consider opening a pull request if you would like to suggest a feature or contribute to this project. Pull
request(s) will be approved in a timely manner. :)

- Make sure to report bugs that may significantly affect research.
- Open a pull request with your changes, following these guidelines.
- Follow these guidelines and your pull request(s) shall be accepted.

## License(s)

GNU General Public License v3.0

## Citation(s)

If you plan on using SSMS Package for future publications, please include the following citation(s):

> Andrew, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data. http://www.bioinformatics.babraham.ac.uk/projects/fastqc/

> Bolger, A. M., Lohse, M., & Usadel, B. (2014). Trimmomatic: A flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114–2120. https://doi.org/10.1093/bioinformatics/btu170

> Bolyen, E., Rideout, J. R., Dillon, M. R., Bokulich, N. A., Abnet, C. C., Al-Ghalith, G. A., Alexander, H., Alm, E. J., Arumugam, M., Asnicar, F., Bai, Y., Bisanz, J. E., Bittinger, K., Brejnrod, A., Brislawn, C. J., Brown, C. T., Callahan, B. J., Caraballo-Rodríguez, A. M., Chase, J., … Caporaso, J. G. (2019). Reproducible, interactive, scalable and extensible microbiome data science using QIIME 2. Nature Biotechnology, 37(8), 852–857. https://doi.org/10.1038/s41587-019-0209-9

> Ewels, P., Magnusson, M., Lundin, S., & Käller, M. (2016). MultiQC: Summarize analysis results for multiple tools and samples in a single report. Bioinformatics, 32(19), 3047–3048. https://doi.org/10.1093/bioinformatics/btw354
Langmead, B., & Salzberg, S. L. (2012). Fast gapped-read alignment with Bowtie 2. Nature Methods, 9(4), 357–359. https://doi.org/10.1038/nmeth.1923

> Lu, J., Breitwieser, F. P., Thielen, P., & Salzberg, S. L. (2017). Bracken: Estimating species abundance in metagenomics data. PeerJ Computer Science, 3, e104. https://doi.org/10.7717/peerj-cs.104

> McIver, L. J., Abu-Ali, G., Franzosa, E. A., Schwager, R., Morgan, X. C., Waldron, L., Segata, N., & Huttenhower, C. (2018). bioBakery: A meta'omic analysis environment. Bioinformatics (Oxford, England), 34(7), 1235–1237. https://doi.org/10.1093/bioinformatics/btx754

> **Testo, G., Lee, K., Gillece, J., Cope, E. (2022).** SSMS Package (simplified metagenomic workflow). The Pathogen & Microbiome Institute. https://github.com/metro1102/ssms-package

> Wood, D. E., Lu, J., & Langmead, B. (2019). Improved metagenomic analysis with Kraken 2. Genome Biology, 20(1), 257. https://doi.org/10.1186/s13059-019-1891-0
